Lib/inspect.py

```diff
@@ -3285,20 +3286,21 @@ def _bind(self, args, kwargs, *, partial=False):
 
             else:
                 if param.kind == _POSITIONAL_ONLY:
-                    # This should never happen in case of a properly built
-                    # Signature object (but let's have this check here
-                    # to ensure correct behaviour just in case)
-                    raise TypeError('{arg!r} parameter is positional only, '
-                                    'but was passed as a keyword'. \
-                                    format(arg=param.name))
-
-                arguments[param_name] = arg_val
+                    # Restore the param in case there is a kwargs_param
+                    kwargs[param_name] = arg_val
+                else:
+                    arguments[param_name] = arg_val
```

Comment:
All `.call` tests do not test `.arguments` property of the `BoundSignature` object. Please, add test cases for that as well: with and without kwargs. Since your code touches this place.

Reply:
A benefit of the refactor I just mentioned (thanks for the suggestion!) is that this is no longer part of my diff.